File: packages/shared-types/src/inventory.ts
export type InventoryStatus = "estimated" | "estimating" | "low_stock" | "no_records";

export type InventoryState = {
  id: string;
  householdItemId: string;
  estimatedRemainingQty: number;
  estimatedRemainingDays: number;
  dailyConsumptionRate: number;
  confidenceScore: number;
  belowSafetyStock: boolean;
  calcReason: string;
  lastRecalcAt: string;
};

export type ItemInventorySummary = {
  estimatedRemainingDays: number | null;
  inventoryStatus: InventoryStatus;
  lastPurchasedAt: string | null;
};

File: apps/api/app/schemas/inventory.py
from __future__ import annotations

from datetime import datetime

from pydantic import BaseModel, ConfigDict


class InventoryStateResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: str
    household_item_id: str
    estimated_remaining_qty: float
    estimated_remaining_days: int
    daily_consumption_rate: float
    confidence_score: float
    below_safety_stock: bool
    calc_reason: str
    last_recalc_at: datetime
